haskell/src/Monocle/Api/Server.hs

```diff
@@ -114,43 +115,60 @@ userGroupGet request = do
 pattern ProjectEntity project =
   Just (CrawlerPB.Entity (Just (CrawlerPB.EntityEntityProjectName project)))
 
+pattern OrganizationEntity organization =
+  Just (CrawlerPB.Entity (Just (CrawlerPB.EntityEntityOrganizationName organization)))
+
 toEntity :: Maybe CrawlerPB.Entity -> Entity
 toEntity entityPB = case entityPB of
   ProjectEntity projectName -> Project $ toStrict projectName
+  OrganizationEntity organizationName -> Organization $ toStrict organizationName
   otherEntity -> error $ "Unknown Entity type: " <> show otherEntity
 
 -- | /crawler/add endpoint
 crawlerAddDoc :: CrawlerPB.AddDocRequest -> AppM CrawlerPB.AddDocResponse
 crawlerAddDoc request = do
   Env {tenants = tenants} <- ask
-  let (CrawlerPB.AddDocRequest indexName crawlerName apiKey entity changes events) = request
+  let ( CrawlerPB.AddDocRequest
+          indexName
+          crawlerName
+          apiKey
+          entity
+          changes
+          events
+          projects
+        ) = request
 
   let requestE = do
         index <-
           Config.lookupTenant tenants (toStrict indexName)
             `orDie` CrawlerPB.AddDocErrorAddUnknownIndex
 
-        _crawler <-
+        crawler <-
           Config.lookupCrawler index (toStrict crawlerName)
             `orDie` CrawlerPB.AddDocErrorAddUnknownCrawler
 
         when
           (Config.crawlers_api_key index /= Just (toStrict apiKey))
           (Left CrawlerPB.AddDocErrorAddUnknownApiKey)
 
-        pure index
+        pure (index, crawler)
 
   case requestE of
-    Right index -> runTenantM index $ case toEntity entity of
+    Right (index, crawler) -> runTenantM index $ case toEntity entity of
       Project _ -> addChanges crawlerName changes events
-      Organization _ -> error "Organization entity not yet supported"
+      Organization organizationName -> addProjects crawler organizationName projects
     Left err -> pure $ toErrorResponse err
   where
     addChanges crawlerName changes events = do
       monocleLogEvent $ AddingChange crawlerName (length changes) (length events)
       I.indexChanges (map I.toELKChange $ toList changes)
       I.indexEvents (map I.toELKChangeEvent $ toList events)
       pure $ CrawlerPB.AddDocResponse Nothing
+    addProjects crawler organizationName projects = do
+      monocleLogEvent $ AddingProject (getWorkerName crawler) organizationName (length projects)
+      I.initCrawlerEntities (Project <$> projectNames projects) crawler
```

Comment:
Why this need the `crawler` and shouldn't the function be named `I.indexProject` ?

Reply:
Here we are not indexing projects but instead we set the for each project a crawler metadata entity. `setOrUpdateLastUpdated` function is used to perform this operation and need the crawler.